test/controllers/manage/agreements_controller_test.rb

```diff
@@ -0,0 +1,224 @@
+      post :create, params: { agreement: { name: "Fun Agreement", agreement_fun: "https://bar.edu" } }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+
+    should "not allow access to manage_agreements#update" do
+      patch :update, params: { id: @agreement, agreement: { name: "Not fun agreement" } }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+
+    should "not allow access to manage_agreements#destroy" do
+      patch :destroy, params: { id: @agreement }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+  end
+
+  context "while authenticated as an organizer" do
+    setup do
+      @user = create(:organizer)
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      sign_in @user
+    end
+
+    should "allow access to manage_agreements#index" do
+      get :index
+      assert_response :redirect
+    end
+
+    should "not allow access to manage_agreements#new" do
+      get :new
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+
+    should "not allow access to manage_agreements#edit" do
+      get :edit, params: { id: @agreement }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+
+    should "not allow access to manage_agreements#create" do
+      post :create, params: { agreement: { name: "Fun Agreement", agreement_fun: "https://bar.edu" } }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+
+    should "not allow access to manage_agreements#update" do
+      patch :update, params: { id: @agreement, agreement: { name: "Not fun agreement" } }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+
+    should "not allow access to manage_agreements#destroy" do
+      patch :destroy, params: { id: @agreement }
+      assert_response :redirect
+      assert_redirected_to manage_agreements_path
+    end
+  end
+
+  context "while authenticated as a director" do
+    setup do
+      @user = create(:director)
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      sign_in @user
+    end
+
+    should "allow access to manage_agreements#index" do
+      get :index
+      assert_response :success
+    end
+
+    should "allow access to manage_agreements#new" do
+      get :new
+      assert_response :success
+    end
+
+    should "create a new agreement" do
+      post :create, params: { agreement: { name: "Fun Agreement", agreement_url: "https://foo.com" } }
+      assert_response :redirect
+    end
+
+    should "allow access to manage_agreements#edit" do
+      get :edit, params: { id: @agreement }
+      assert_response :success
+    end
+
+    should "update agreement" do
+      patch :update, params: { id: @agreement, agreement: { name: "New agreement Name" } }
```

Comment:
Not checking if new agreement name is actually set. Also, updates appear to be broken -- just keeps me on the page while stating the error `Agreement URL must start with http:// or https://`.

Reply:
#430